### PR TITLE
make DB creation optional

### PR DESCRIPTION
**Pull request by thattommyhall**

### CODE REVIEW

**Review by alanshaw — CHANGES_REQUESTED** [review]
Code changes LGTM but I don't understand why necessary? Does `IF NOT EXISTS` not work for this?

We need changes to the k8s config for each node https://github.com/libp2p/hydra-booster/tree/master/k8s to add this env var.

**Comment by thattommyhall** [comment]
We are now certain that (on DO at least) the `CREATE INDEX` command waits for about a day to return that the index already exists.

**Comment by thattommyhall** [comment]
Note that the prod deployment is now pointed at a Docker image I made, we should fixup asap 

```
image: docker.io/thattommyhall/hydra-booster:make-db-creation-optional
```

File: k8s/alasybil.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ spec:
               name: secrets
               key: db
         image: docker.io/libp2p/hydra-booster
+        - name: HYDRA_DISABLE_DBCREATE
+          value: "true"
```
Comment (alanshaw) [inline k8s/alasybil.yaml:40]: These env vars are not in the right place - need to be under the `env` prop.

File: datastore/postgres.go [hunk 2]
```diff
@@ -11,24 +12,29 @@ import (
...
+		}
+		fmt.Fprintf(os.Stderr, "Creating Index\n")
+		_, err = pool.Exec(ctx, fmt.Sprintf("CREATE INDEX IF NOT EXISTS %s_key_text_pattern_ops_idx ON %s (key text_pattern_ops)", tableName, tableName))
+		if err != nil {
+			return nil, err
+		}
+		pool.Close()
 	}
-	pool.Close()
+	fmt.Fprintf(os.Stderr, "Connecting to Database\n")
```
Comment (alanshaw) [inline datastore/postgres.go:37]: Shall we remove these `fmt.Fprintf` logs or do you think they're useful?

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: k8s/alasybil.yaml (hunk 1 above)
Comment (thattommyhall) [reply k8s/alasybil.yaml:40]: doh, It was because I didn't commit my image change

File: datastore/postgres.go (hunk 2 above)
Comment (thattommyhall) [reply datastore/postgres.go:37]: It was hanging at the 2nd of the three things we do here, nice to know where it is I think

**Review by alanshaw — APPROVED** [review]

**Pull request merged**